Problem with read_local_config test

### What happened?

After updating to v0.8.1 I ran the tests and test_read_config_check_old_stores failed at L48:
`"We've only detected old (non-Zarr) object stores. Please update your configuration to add a new path."`
This seems weird, because if I understand correctly (which I might not) the store referred to in the mock config file shouldn't have been created yet. Could it be that there's some left over test store in the temp directory that's confusing things here?

### What did you expect to happen?

I hoped the tests would all pass!

### Minimal Complete Verifiable Example

_No response_

### Relevant log output

```Python
=================================================================================== FAILURES ===================================================================================
______________________________________________________________________ test_read_config_check_old_stores _______________________________________________________________________

mock_get_user_config_path = None, write_mock_config = None, caplog = <_pytest.logging.LogCaptureFixture object at 0x7f6ae82d1b20>

    def test_read_config_check_old_stores(
        mock_get_user_config_path, write_mock_config, caplog
    ):
        """This tests the read_local_config function when the user has an old store in their config file.
        This test and the _check_valid_store function may be removed once the move to the new store setup is complete.
        """
>       config = read_local_config()

util/test_user.py:48: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def read_local_config() -> Dict:
        """Reads the local config file.
    
        Returns:
            dict: OpenGHG configurations
        """
        config_path = get_user_config_path()
    
        if not config_path.exists():
            try:
                _migrate_config()
            except FileNotFoundError as e:
                raise ConfigFileError(
                    "Unable to read configuration file, please see the installation instructions \
                    or run openghg --quickstart"
                ) from e
    
        config: Dict = toml.loads(config_path.read_text())
    
        try:
            _ = config["object_store"]["user"]
        except KeyError:
            raise ConfigFileError(
                "Invalid config file detected, please please see the installation instructions \
                    or run openghg --quickstart"
            )
    
        # Check see is the store uses the new zarr storage format
        # for OpenGHG >= 0.8.0
        valid_stores = {}
        for name, store_data in config["object_store"].items():
            store_path = Path(store_data["path"])
            # If it doesn't exist or its empty then we expect it to be created / populated
            if not store_path.exists() or not any(store_path.iterdir()):
                valid_stores[name] = store_data
            # Otherwise we check an existing store to see if it's the correct format
            else:
                if _check_valid_store(store_path):
                    valid_stores[name] = store_data
                else:
                    logger.warning(
                        f"Object store {name} does not use the new Zarr storage format and will be ignored."
                    )
    
        if not valid_stores:
>           raise ConfigFileError(
                "We've only detected old (non-Zarr) object stores. Please update your configuration to add a new path."
            )
E           openghg.types._errors.ConfigFileError: We've only detected old (non-Zarr) object stores. Please update your configuration to add a new path.

../openghg/util/_user.py:295: ConfigFileError
```


### Anything else we need to know?

_No response_

### Environment

<details>
Python 3.12.3
OpenGHG 0.8.1


</details>

--- a/tests/util/test_user.py
+++ b/tests/util/test_user.py
@@ -8,14 +8,8 @@ from openghg.util import check_config, create_config, read_local_config
 
 
 @pytest.fixture
-def tmp_config_path(tmpdir):
-    return Path(tmpdir).joinpath("config_folder").joinpath("mock_config.conf")
-
-
-@pytest.fixture
-def tmp_home_path(tmpdir):
-    return Path(tmpdir).joinpath("tmp_home_path")
-
+def tmp_config_path(tmp_path):
+    return tmp_path.joinpath("config_folder").joinpath("mock_config.conf")
 
 @pytest.fixture
 def mock_get_user_config_path(tmp_config_path, mocker):
@@ -24,14 +18,14 @@ def mock_get_user_config_path(tmp_config_path, mocker):
 
 
 @pytest.fixture
-def write_mock_config(tmpdir, tmp_config_path):
+def write_mock_config(tmp_path, tmp_config_path):
     mock_uuid = "179dcd5f-d5bb-439d-a3c2-9f690ac6d3b8"
-    mock_path = Path(tmpdir).joinpath("mock_store")
-    mock_shared_path = Path(tmpdir).joinpath("mock_shared_store")
+    mock_path = tmp_path.joinpath("mock_store")
+    mock_shared_path = tmp_path.joinpath("mock_shared_store")
     mock_conf = {
         "object_store": {
-            "user": {"path": mock_path, "permissions": "rw"},
-            "shared": {"path": mock_shared_path, "permission": "rw"},
+            "user": {"path": str(mock_path), "permissions": "rw"},
+            "shared": {"path": str(mock_shared_path), "permission": "rw"},
         },
         "user_id": mock_uuid,
     }
@@ -81,9 +75,9 @@ def test_read_config_check_old_stores(
         read_local_config()
 
 
-def test_create_config(monkeypatch, mocker, tmpdir):
-    monkeypatch.setenv("HOME", str(tmpdir))
-    mock_config_path = Path(tmpdir).joinpath("mock_config.conf")
+def test_create_config(monkeypatch, mocker, tmp_path):
+    monkeypatch.setenv("HOME", str(tmp_path))
+    mock_config_path = tmp_path.joinpath("mock_config.conf")
 
     mocker.patch("openghg.util._user.get_user_config_path", return_value=mock_config_path)
     mock_uuids = [f"test-uuid-{x}" for x in range(100, 110)]
@@ -120,9 +114,9 @@ def test_create_config_migrate(mocker, monkeypatch, tmp_path, caplog):
     assert mock_new_config_path.read_text() == mock_config_content
 
 
-def test_check_config(mocker, caplog, monkeypatch, tmpdir):
-    monkeypatch.setenv("HOME", str(tmpdir))
-    mock_config_path = Path(tmpdir).joinpath("mock_config.conf")
+def test_check_config(mocker, caplog, monkeypatch, tmp_path):
+    monkeypatch.setenv("HOME", str(tmp_path))
+    mock_config_path = tmp_path.joinpath("mock_config.conf")
 
     mocker.patch("openghg.util._user.get_user_config_path", return_value=mock_config_path)
 
@@ -145,13 +139,13 @@ def test_check_config(mocker, caplog, monkeypatch, tmpdir):
     assert " /tmp/mock_store does not exist but will be created." in caplog.text
 
 
-def test_create_config_duplicates(monkeypatch, mocker, tmpdir, capsys):
+def test_create_config_duplicates(monkeypatch, mocker, tmp_path, capsys):
     """
     Test simulates input values submitted after invoking the create_config method. It verifies if the value error is raised for duplicate store names and store paths.
     """
 
-    monkeypatch.setenv("HOME", str(tmpdir))
-    mock_config_path = Path(tmpdir).joinpath("mock_config.conf")
+    monkeypatch.setenv("HOME", str(tmp_path))
+    mock_config_path = tmp_path.joinpath("mock_config.conf")
 
     mocker.patch("openghg.util._user.get_user_config_path", return_value=mock_config_path)
     mock_uuids = [f"test-uuid-{x}" for x in range(100, 110)]

